opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -86,94 +90,142 @@ message InstrumentationLibraryMetrics {
-    // (usually things like request durations and response sizes). While it
-    // also provides a total count of observations and a sum of all observed
-    // values, it calculates configurable percentiles over a sliding time
-    // window.
-    // Corresponding values are stored in SummaryDataPoint.
-    SUMMARY = 6;
+    // Summary implies that the value is found in
+    // Metric.summary_data_points[].summary.
+    SUMMARY = 4;
+
+    // TODO(#159): add RAW_INT64 and RAW_DOUBLE value types.
   }
 
-  // type is the type of values this metric has.
-  Type type = 4;
+  // ValueType is the type of values this metric has.
+  ValueType value_type = 4;
+
+  // KindElement contains a set of bit masks used to construct Kind
+  // enum values.  There are 7 bits used presently, broken into
+  // groups:
+  //
+  // "Temporality" is the temporal quality of a metric, indicating how
+  // values relate to the time interval over which they are reported.
+  // One of the 3 Temporality values MUST be set: CUMULATIVE, DELTA,
+  // or INSTANTANEOUS.
+  //
+  // "Structure" is the structural quality of a metric, indicating
+  // whether metric data describes a sum of measurements (ADDING) or a
+  // collection of individual measurements (GROUPING).  Value types
+  // may be interpreted differently depending on Structure.  For
+  // example, a Histogram DataPoint may be computed for a Counter
+  // (ADDING) instrument or a ValueRecorder (GROUPING) instrument, and
+  // the Sum and Count of the resulting aggregation may be interpreted
+  // differently depending on structure.
+  //
+  // MONOTONIC is a boolean option that may be applied to
+  // ADDING-structure metrics.
+  //
+  // ASYNCHRONOUS is a boolean option that may be applied to all
+  // metrics.  When set, an ASYNCHRONOUS kind indicates that the
+  // corresponding measurement was made through a callback called by
+  // the SDK.  TimeUnixNano is deliberate, not the result of an
+  // application-level event.
+  //
+  // When measurements originate from an OpenTelemetry API, the
+  // ASYNCHRONOUS kind allows the interpreter to recognize values that
+  // were emitted by the same callback invokation, as identified by
+  // equal an Resource and TimeUnixNano.  OpenTelemetry callbacks are
+  // permitted to record at most one value per label set under the
+  // specification.  This property can be safetly disregarded and not
+  // set when importing data from other sources, but when set its
+  // presence is always meaningful.
+  //
+  // ASYNCHRONOUS means that:
+  // - The count of events is a measure of the rate of SDK collection
+  //   times the aggregated cardinality, therefore cannot be used
+  //   to extrapolate application-specific rate information.
+  // - Measurements with the same Resource and TimeUnixNano
+  //   form a meaningful ratio, and the set of measurements at this
+  //   instant are coherent, resulting from a single callback.
+  // - Trace context MUST not be set in exemplars.
```

Comment:
This probably does not have exemplars at all, correct?

Reply:
Not exemplars, but associated RawValues makes sense.

Reply:
What is the difference between Exemplar and RawValue?

Reply:
```
  // - Measurements are not traced, by definition, and SHOULD NOT have
  //   associated trace context (e.g., no exemplars with trace_id,
  //   span_id).
```